src/views/CreateView.vue

```diff
@@ -263,6 +265,30 @@ export default {
       }
 
       return selectedExercises
+    },
+    errors() {
+      return {
+        exerciseName: {
+          conditions: this.isFormSubmitted && this.workoutName === '',
+          text: 'The "Workout name" value is required.'
+        },
+        sets: {
+          conditions: this.isFormSubmitted && this.exerciseData.sets === '',
+          text: 'The "Sets" value is required.'
+        },
+        reps: {
+          conditions: this.isFormSubmitted && this.exerciseData.reps === '',
+          text: 'The "Reps" value is required.'
+        },
+        weight: {
+          conditions: this.isFormSubmitted && (this.exerciseData.weight === '' || this.exerciseData.unit === ''),
+          text: 'The "Weight" value and its unit are required.'
+        },
```

Comment:
I believe unit should has a separate error message. That could actually help you making error message reusable:

```js
errors() {
      function getErrorMessage(name) {
        return `The "${ name }" value is required.`
      }
      return {
        exerciseName: {
          conditions: this.isFormSubmitted && this.workoutName === '',
          text: getErrorMessage("Workout name"),
        },
        sets: {
          conditions: this.isFormSubmitted && this.exerciseData.sets === '',
          text: getErrorMessage('Sets')
        },
//...
```

Reply:
I believed so as well but the thing is I'm unable to position this message under the unit inputs due to at least two reasons, which I find insignificant for this matter but they are related to the way the elements are positioned now. And since I can't use a separate error message for those inputs I decided to check the the weight and units together and put one message if at least one of them isn't right.

Regarding the code you provided, I'm sorry but I actually didn't notice it before pushing the current code 🙈, so I will try to apply this in my next commit.

Reply:
Sure thing 👌

Reply:
**Implemented your code**

I had no idea I can pass functions into a computed property. I believed I could only return data. Now, I'll remember I can use it like that 🙃

Reply:
oh yes you can. Computed properties are functions, right? In the end of the day it's just JS